pkg/volmount/getvolumenameforvolumemountpoint.go

```diff
@@ -0,0 +1,33 @@
+package volmount
+
+import (
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// GetVolumeNameForVolumeMountPoint returns a volume path (in format '\\?\Volume{GUID}'
+// for the volume mounted at targetPath.
+// https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw
+func GetVolumeNameForVolumeMountPoint(targetPath string) (string, error) {
+	// Must end in a backslash
+	slashedTarget := filepath.Clean(targetPath)
+	if slashedTarget[len(slashedTarget)-1] != filepath.Separator {
+		slashedTarget = slashedTarget + string(filepath.Separator)
+	}
+
+	targetP, err := windows.UTF16PtrFromString(slashedTarget)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to utf16-ise %s", slashedTarget)
+	}
+
+	bufferlength := uint32(50) // "A reasonable size for the buffer" per the documentation.
```

Comment:
I think you're thinking of the next line, which creates a `uint16` array of this length? This is a `uint32` because we also pass the `bufferLength` to the Win32 API call, and it requires a `uint32`.

Or am I misunderstanding, and you are suggesting I should have
```
var buffer [50]uint16

if err = windows.GetVolumeNameForVolumeMountPoint(targetP, &buffer[0], len(buffer)); err != nil {
```
and then take a slice of `buffer` when I call `windows.UTF16ToString` later? I thought that's what `make` was effectively doing for me anyway.

Reply:
Never mind I was reading it wrong.